test/ruby/test_yjit.rb

```diff
@@ -708,7 +869,9 @@ def collect_insns(iseq)
       recorded_exits = recorded_exits.reject { |k, v| v == 0 }
 
       recorded_exits.transform_keys! { |k| k.to_s.gsub("exit_", "").to_sym }
-      if exits != :any && exits != recorded_exits
+      if exits != :any &&
+        exits != recorded_exits &&
+        !exits.all? { |k, v| v === recorded_exits[k] } # triple-equal allows range comparison
```

Comment:
Should have a comment above this if-statement explaining what is being checked, what does `!exits.all? { |k, v| v === recorded_exits[k] }` do?

Reply:
There's a comment to the right that says "triple-equal allows range comparison." But I'll expand it and move it above the "if".